### PR TITLE
[FEATURE] import export synonyms and stop words

### PR DESCRIPTION
**Pull request by sfsmfc**
Implements the import and export for synonyms and stopwords.

Fixes: #1135

### CODE REVIEW

File: Classes/Utility/ManagedResourcesUtility.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * Class ManagedResourcesUtility
+ */
+class ManagedResourcesUtility
+{
+
+    /**
+     *
```
Comment (timohund) [inline Classes/Utility/ManagedResourcesUtility.php]: I think we should add a desciption here or remove the empty lines

File: Classes/Controller/Backend/Search/CoreOptimizationModuleController.php [hunk 2]
```diff
@@ -141,28 +251,21 @@ public function deleteSynonymsAction($baseWord)
      * Saves the edited stop word list to Solr
      *
      * @param string $stopWords
+     * @param int $replaceStopwords
```
Comment (timohund) [inline Classes/Controller/Backend/Search/CoreOptimizationModuleController.php]: I would propose to use a boolean value here.

File: Classes/Controller/Backend/Search/CoreOptimizationModuleController.php [hunk 3]
```diff
@@ -111,6 +117,110 @@ public function addSynonymsAction(string $baseWord, string $synonyms)
...
+     */
+    public function exportSynonymsAction($fileFormat = 'txt')
+    {
+        $synonyms = $this->selectedSolrCoreConnection->getSynonyms();
+        return $this->exportFile(ManagedResourcesUtility::exportSynonymsToTxt($synonyms), 'synonyms', $fileFormat);
+    }
+
+    /**
+     * @param array $synonymFileUpload
+     * @param int $overrideExisting
```
Comment (timohund) [inline Classes/Controller/Backend/Search/CoreOptimizationModuleController.php]: I would propose to use a boolean value here

File: Classes/Controller/Backend/Search/CoreOptimizationModuleController.php [hunk 4]
```diff
@@ -82,9 +84,10 @@ public function indexAction()
      *
      * @param string $baseWord
      * @param string $synonyms
+     * @param int $overrideExisting
```
Comment (timohund) [inline Classes/Controller/Backend/Search/CoreOptimizationModuleController.php]: I would propose to use a bool value here

**Comment by sfsmfc** [comment]
done

**Comment by sfsmfc** [comment]
done

File: Classes/Controller/Backend/Search/CoreOptimizationModuleController.php [hunk 5]
```diff
@@ -111,6 +117,105 @@ public function addSynonymsAction(string $baseWord, string $synonyms)
...
+        foreach ($fileLines as $baseWord => $synonyms) {
+            if (!isset($baseWord) || empty($synonyms)) {
+                continue;
+            }
+            $this->deleteExistingSynonym($overrideExisting, $deleteSynonymsBefore, $baseWord);
+            $this->selectedSolrCoreConnection->addSynonym(
+                $baseWord,
+                $synonyms
+            );
+            $this->selectedSolrCoreConnection->reloadCore();
```
Comment (timohund) [inline Classes/Controller/Backend/Search/CoreOptimizationModuleController.php]: I would propose to do the reload outside of the loop.

Comment (sfsmfc) [inline Classes/Controller/Backend/Search/CoreOptimizationModuleController.php]: The core will only reload once after adding all synonyms.

**Pull request merged**